src/app/content/components/Page/highlightManager.ts

```diff
@@ -163,10 +178,22 @@ export default (container: HTMLElement, getProp: () => HighlightProp) => {
         ;
 
       highlighter.clearFocus();
-      const focusedId = getProp().focused;
+      const { scrollTarget, focus, focused: focusedId, highlightsLoaded, loggedOut, page } = getProp();
+      const highlightScrollTarget = scrollTarget && scrollTarget.type === 'highlight' ? scrollTarget : null;
       const focused = focusedId && highlighter.getHighlight(focusedId);
-      if (focused) {
-        focused.focus();
+      const scrollTargetHighlight = highlightScrollTarget && highlighter.getHighlight(highlightScrollTarget.id);
+      const toFocus = scrollTargetHighlight || focused;
```

Comment:
i don't think we can set the priority like this, we need to do it in a way where when one of these is changed we process it